scalafix-core/shared/src/main/scala/scalafix/patch/Patch.scala

```diff
@@ -122,9 +122,6 @@ object Patch {
       patches: Map[RuleName, Patch],
       ctx: RuleCtx): List[LintMessage] = {
 
-    // TODO(olaf): Remove this cast
-    val ctxImpl = ctx.asInstanceOf[RuleCtxImpl]
```

Comment:
I'm eager to get rid of this cast but this change seems unrelated to the PR.

Reply:
Don't get me wrong, this change is nice and we can keep it here. I feel it could have have been a separate PR to make this PR easier to review. I admit I'm personally bad at keeping PRs focused on one change, so I promise to try and do better myself.